### PR TITLE
basic security with in-memory users ready

### PR DESCRIPTION
**Pull request by dovantuan97**

### CODE REVIEW

File: src/main/java/com/example/todoz/models/Task.java [hunk 1]
```diff
@@ -41,38 +41,36 @@ public Task() {
...
             return null;
         }
     }
 
-    public String getDueDateDayOfWeek(){
+    public String getDueDateDayOfWeek() {
         return this.dueDate.getDayOfWeek().getDisplayName(TextStyle.SHORT, Locale.getDefault());
     }
 
     public Object getRemainingDays() {
```
Comment (ales) [inline src/main/java/com/example/todoz/models/Task.java:65]: Better String than Object?

File: src/main/java/com/example/todoz/services/TaskService.java [hunk 2]
```diff
@@ -75,24 +70,22 @@ public List<Task> findTasksForNextWeek() {
...
+                .toList();
     }
 
     public List<Task> findLongTerm() {
         return taskRepo.findAll()
                 .stream()
-                .filter(t -> t.getDueDateWeek() != null
-                        && t.getDueDateWeek() > Week.getCurrentWeekNumber())
+                .filter(t -> t.getDueDateWeek() != null && t.getDueDateWeek() > Week.getCurrentWeekNumber())
                 .sorted(Comparator.comparing(Task::getDueDate))
```
Comment (ales) [inline src/main/java/com/example/todoz/services/TaskService.java:80]: In future, we better do this on db level -- findAllOrderBy-something

**Review by ales — APPROVED** [review]
Mrkni na komenty, jinak gg, hezky!

**Comment by ales** [comment]
Ty komenty se ovsem netykaji tvych zmen - za to pardon :) chtel jsem jen, abys*te o tom vedel*i

**Review by BelovUser — APPROVED** [review]

**Review by Davias97 — APPROVED** [review]

File: src/main/java/com/example/todoz/models/Task.java (hunk 1 above)
Comment (dovantuan97) [reply src/main/java/com/example/todoz/models/Task.java:65]: @BelovUser autogenerated right?

**Comment by ales** [comment]
bez prihlaseni neni pristup k assetum, proto login page vypada takto

Screenshot 2023-12-16 at 21 29 05

**Review by yenik — APPROVED** [review]

File: src/main/java/com/example/todoz/security/SecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+import org.springframework.security.web.SecurityFilterChain;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig {
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http.authorizeHttpRequests(request -> request.requestMatchers("/login")
```
Comment (ales) [inline src/main/java/com/example/todoz/security/SecurityConfig.java:22]: requestMatchers("/login", "/*.css", "/img/*")

**Pull request merged**